### PR TITLE
Ensure MeterListener.RecordObservableInstruments invoking all instruments callbacks

### PR DESCRIPTION
**Pull request by tarekgh**
When having a MeterListener listening to multiple observable instruments and this listener is calling back all listening to instruments to collect instrumentation data; if one of the callbacks throw exception, we used to let this exception propagate. The remaining instruments callbacks will not get a chance to execute. This change is catching any exception thrown from the instrument callbacks and continue executing calling the remaining instruments. At the end of the operation, we'll throw AggregationException containing the list of all exceptions thrown during the operation.

### CODE REVIEW

**Comment by tarekgh** [comment]
@noahfalk @cijothomas could you please have a look?

CC @shirhatti @reyang

**Review by noahfalk — APPROVED** [review]
LGTM! Once it is merged I can also confirm the errors are reported reasonably in dotnet-counters

cc @hoyosjs

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricsTests.cs
Comment (cijothomas) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricsTests.cs:190]: consider adding a test case where some callbacks throw, and some does not throw, and validate that SetMeasurementEventCallback is called for the ones that did not throw.

**Review by cijothomas — APPROVED** [review]
LGTM.
Added a small suggestion for an additional test case

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricsTests.cs
Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricsTests.cs:199]: I did. Thanks!

**Comment by tarekgh** [comment]
The CI failure is unrelated and tracked with the issue https://github.com/dotnet/runtime/issues/56190

**Pull request merged**
